src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -255,4 +265,95 @@ private static Stream<Arguments> invalidNames() {
         Arguments.of("Te*st")
     );
   }
+
+  @Test
+  public void requestRoleChange_valid() throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException,
+      AuthenticationProviderMappingException, IncorrectCredentialsProvidedException,
+      AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException,
+      MFARequiredButNotConfiguredException {
+    LoginResult currentlyStudentLogin = loginAs(httpSession, ITConstants.TEST_STUDENT_EMAIL, ITConstants.TEST_STUDENT_PASSWORD);
+    HttpServletRequest upgradeRequest = createRequestWithCookies(new Cookie[] {currentlyStudentLogin.cookie});
+    replay(upgradeRequest);
+    Map<String, String> requestDetails = Map.of(
+        "verificationDetails", "school staff url",
+        "otherDetails", "more information"
+    );
+
+    Response upgradeResponse = usersFacade.requestRoleChange(upgradeRequest, requestDetails);
+
+    assertEquals(Response.Status.OK.getStatusCode(), upgradeResponse.getStatus());
```

Comment:
Checking the state of the flag should be pretty simple. Not so sure about the email though.
As previous discussed, I don't think the email provider actually exists in the integration tests so we're limited to checking if the appropriate function is called and that might be tricky with how the integration tests are built. Need to look into that and see how many objects would need rebuilding for the specific test...

Reply:
Yeah, the email could be tricky... It may be worth spending a few minutes checking if there is anything quick and easy that can be done in order to test it, like mocking the email manager (but don't use PowerMock, please). If it's not easy, then we should at the very least test that the teacher flag has changed.